instrumentation/openai-client-instrumentation/src/test/java/co/elastic/otel/openai/ValAssert.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.otel.openai;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.common.KeyValue;
+import io.opentelemetry.api.common.Value;
+import io.opentelemetry.api.common.ValueType;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public interface ValAssert extends Consumer<Value<?>>, BiConsumer<String, Value<?>> {
```

Comment:
I know this might require a bit of effort, but it might be better to do a proper assertj extension as we have in our classic agent, the downside however is having to rewrite assertions using a different API. Given it's about otel types, those could even be contributed upstream and reused.

Reply:
I agree that this would be the ideal solution, but also a lot of effort to get readable assertions. Are you okay with keeping it as is for now?

Reply:
Yes, that's fine. Contributing this upstream would probably be a better choice anyway, then reusing them here would be likely easier.